Made progress in expanding AT and DW commands, started working on TCP…

DrPitre opened the pull request:
… Commands

Commit:
Made progress in expanding AT and DW commands, started working on TCP Commands

tlindner commented on source/DriveWireServerModel.m:
```diff
@@ -1606,6 +1612,7 @@ - (NSUInteger)OP_SERREADM:(NSData *)data;
         NSData *dataToRead = [channel getNumberOfBytes:bytesToRead];
 
         [portDelegate writeData:dataToRead];
+        NSLog(@"%@", dataToRead); 
```
Debugging Log?

DrPitre replied:
Totally bogus. Removing.

tlindner commented on source/VirtualSerialChannel+DWCommands.m:
```diff
@@ -88,43 +89,93 @@ - (NSError *)handleDWSERVERLIST:(NSArray *)array;
+
+        NSFileHandle *file = [pipe fileHandleForReading];
+        
+        [task launch];
```
This is synchronous, right? Should there be a hard timeout on the launched process?

DrPitre replied:
Yes, this needs a "cut off". I'll look at implementation.

tlindner commented on source/VirtualSerialChannel.m:
```diff
@@ -159,19 +177,37 @@ - (void)putBytes:(u_char *)bytes length:(NSUInteger)length;
+    {
+        if ([result characterAtIndex:i] == '\b' && i > 0)
+        {
+            NSString *beforeBackspace = [result substringToIndex:i - 1];
```
I think this will fail with a \b at the beginning of a line.

Commit:
Fixed issue with backspace processing in special case
Added unit test target, unit test for testing backspace processing
Separated backspace processing into an NSString category
Now there is a 5 second timeout for any task originating from a 'dw server cmd'
Extraneous NSLog removed.
